### PR TITLE
Remove redundant cell cycle code

### PR DESCRIPTION
**Pull request by mumichae**
For some reason, the cell cycle score contains redundant code, which makes it very difficult to debug.

Other issues addressed here:
- make it possible to enforce recomputation of cell cycle scores, even if they exist
- rename 'covariate' to 'variable' to avoid confusion

### CODE REVIEW

File: tests/metrics/test_pcr_metrics.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+from tests.common import *
+import pandas as pd
+
+
+def test_pc_regression(adata):
+    scIB.me.pc_regression(adata.X, adata.obs["batch"])
```
Comment (mbuttner) [inline tests/metrics/test_pcr_metrics.py:6]: What is tested here?

**Review by mbuttner — COMMENTED** [review]
Looks good overall, I did not fully understand all `assert` calls. Is this tested?

File: tests/metrics/test_pcr_metrics.py (hunk 1 above)
Comment (LuckyMD) [reply tests/metrics/test_pcr_metrics.py:6]: implementation... if it doesn't run, there's an errorr

File: tests/metrics/test_silhouette_metrics.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    score = scIB.me.isolated_labels(
+        adata_pca,
+        label_key='celltype',
+        batch_key='batch',
+        embed='X_pca',
+        cluster=False,
+        verbose=True
+    )
+    LOGGER.info(f"score: {score}")
+    assert score <= 1
```
[the remark this answers is not in the document]
Comment (LuckyMD) [reply tests/metrics/test_silhouette_metrics.py:39]: just testing upper and lower boundaries are correct in this test.

**Pull request closed without merging**